### ISSUE TITLE
Apple M1 hardware incompatibility for JNA Native class

### ISSUE THREAD

**Issue by cdsap** [oshi/oshi#1462]
Hi, thanks for this awesome library. 
I'm using the library in [Talaiot](https://github.com/cdsap/Talaiot) and we are getting problems to getMemory with the new M1 mbp. 
The error is:
```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.sun.jna.Native
	at com.sun.jna.Structure.setAlignType(Structure.java:280)
	at com.sun.jna.Structure.<init>(Structure.java:197)
	at com.sun.jna.Structure.<init>(Structure.java:193)
	at com.sun.jna.Structure.<init>(Structure.java:180)
	at com.sun.jna.Structure.<init>(Structure.java:172)
	at com.sun.jna.platform.mac.SystemB$XswUsage.<init>(SystemB.java:365)
	at oshi.hardware.platform.mac.MacGlobalMemory.<init>(MacGlobalMemory.java:51)
	at oshi.hardware.platform.mac.MacHardwareAbstractionLayer.getMemory(MacHardwareAbstractionLayer.java:70)
	at com.cdsap.talaiot.metrics.RamAvailableMetric$1.invoke(SimpleMetrics.kt:56)
```
I updated to the latest version of the library but still similar problem,

Thanks

**Comment by dbwiddis** [comment]
Do you also have the latest version (5.6.0) of the JNA dependency (both `jna` and `jna-platform`)?   [See the steps here](https://github.com/oshi/oshi/blob/master/FAQ.md#how-do-i-resolve-jna-noclassdeffound-errors).

**Comment by dbwiddis** [comment]
Rereading this, I believe this is an issue with JNA.  See https://github.com/java-native-access/jna/pull/1238.  I'm not sure of the current status of that, or if you might be able to help us (I'm also a committer at JNA) figure out the correct resolution.

**Comment by dbwiddis** [comment]
Seems an issue in the JNI world too: https://github.com/openjdk/aarch64-port/pull/3

**Comment by tinder-inakivillar** [comment]
Thanks for the references, will follow up

**Comment by dbwiddis** [comment]
FYI, as a temporary workaround as observed in #1483, OSHI/JNA works (via Rosetta) if you run an x86 JVM.  There also does appear to be some slow progress on updating JNA's `libffi` for aarch64.

**Comment by dbwiddis** [comment]
JNA binaries have been built which should fix this issue, and will be released in JNA 5.7.0 in a few weeks.   Please help test.  See [this post on the jna-users mailing list](https://groups.google.com/g/jna-users/c/UwUa5qWbIco) to download the JARs.

**Comment by dbwiddis** [comment]
JNA 5.7.0 has been released, which should resolve this issue.
